c++/src/kj/compat/http.c++

```diff
@@ -5418,7 +5418,20 @@ public:
   }
 
   kj::Promise<ConnectResponse> connect(kj::StringPtr host, const HttpHeaders& headers) override {
-    KJ_UNIMPLEMENTED("connect() is not implemented for NetworkHttpClient");
+    // We want to connect directly instead of going through a proxy here.
+    // https://github.com/capnproto/capnproto/pull/1454#discussion_r900414879
+    return network.parseAddress(host).then(
+      [&headers](auto address) {
+        return address->connect().then([&headers](auto connection) {
+          return ConnectResponse {
+            .statusCode = 200,
+            .statusText = "OK",
+            .headers = &headers,
+            .connectionOrBody = kj::mv(connection),
+          };
+        });
+      }
+    );
```

Comment:
Lambda body should always be indented one stop more than the surrounding body.

```suggestion
    return network.parseAddress(host)
        .then([&headers](auto address) {
      return address->connect().then([&headers](auto connection) {
        return ConnectResponse {
          .statusCode = 200,
          .statusText = "OK",
          .headers = &headers,
          .connectionOrBody = kj::mv(connection),
        };
      });
    });
```

Reply:
If this seems confusing, consider how you'd indent an `if()` statement where the condition wraps across multiple lines. Do you write:

```
if (really + long + condition ==
  blah + blah) {
    blockContents();
  }
```

Or do you write:

```
if (really + long + condition ==
    blah + blah) {
  blockContents();
}
```